docs/Makefile

```diff
@@ -7,19 +7,108 @@ SPHINXOPTS    ?=
 help:
 	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+	@echo ""
+	@echo "Additional spelling targets:"
+	@echo "  install-spelling-deps  Install pyspelling and aspell dependencies"
+	@echo "  run                    Build docs and run pyspelling"
+	@echo "  spelling               Run pyspelling on existing build"
 
-.PHONY: help Makefile
+.PHONY: help Makefile install-spelling-deps spelling run
+
+# Check if required dependencies are installed, install if missing
+check-spelling-deps:
+	@echo "Checking for required spelling dependencies..."
+	@if ! command -v aspell >/dev/null 2>&1; then \
+		echo "aspell is not installed. Installing dependencies..."; \
+		$(MAKE) install-spelling-deps; \
+	elif ! python3 -c "import pyspelling" 2>/dev/null; then \
+		echo "pyspelling is not installed. Installing dependencies..."; \
+		$(MAKE) install-spelling-deps; \
+	else \
+		echo "All spelling dependencies are installed."; \
+	fi
+
+# Install spelling dependencies
+install-spelling-deps:
+	@echo "Installing spelling dependencies..."
+	@echo "Detecting operating system..."
+	@if command -v apt-get >/dev/null 2>&1; then \
+		echo "Detected Debian/Ubuntu system"; \
+		echo "Installing aspell and aspell-en..."; \
+		sudo apt-get update && sudo apt-get install -y aspell aspell-en; \
+	elif command -v yum >/dev/null 2>&1; then \
+		echo "Detected RHEL/CentOS system"; \
+		echo "Installing aspell and aspell-en..."; \
+		sudo yum install -y aspell aspell-en; \
+	elif command -v dnf >/dev/null 2>&1; then \
+		echo "Detected Fedora system"; \
+		echo "Installing aspell and aspell-en..."; \
+		sudo dnf install -y aspell aspell-en; \
+	elif command -v pacman >/dev/null 2>&1; then \
+		echo "Detected Arch Linux system"; \
+		echo "Installing aspell and aspell-en..."; \
+		sudo pacman -S --noconfirm aspell aspell-en; \
+	elif [ "$$(uname)" = "Darwin" ]; then \
+		echo "Detected macOS system"; \
+		if command -v brew >/dev/null 2>&1; then \
+			echo "Found Homebrew, installing aspell..."; \
+			brew install aspell; \
+		elif command -v port >/dev/null 2>&1; then \
+			echo "Found MacPorts, installing aspell..."; \
+			sudo port install aspell aspell-dict-en; \
+		else \
+			echo "Neither Homebrew nor MacPorts found."; \
+			echo "Please install aspell manually:"; \
+			echo "  - Install Homebrew: https://brew.sh"; \
+			echo "  - Then run: brew install aspell"; \
+			echo "  - Or install MacPorts: https://www.macports.org"; \
+			echo "  - Then run: sudo port install aspell aspell-dict-en"; \
+			exit 1; \
+		fi \
+	else \
+		echo "Could not detect package manager."; \
+		echo "Please manually install aspell and aspell-en for your system."; \
+		echo "Common package names: aspell, aspell-en"; \
+		exit 1; \
+	fi
+	@echo "Installing pyspelling via pip..."
+	@python3 -m pip install pyspelling
+	@echo "Spelling dependencies installed successfully!"
+	@echo "You can now run 'make run' to build docs and check spelling."
```

Comment:
@activus-d where did you source this list from?

Reply:
What list are you referring to here, Storm?

Reply:
'list' -> the list of checks for operating system and package installation commands

Reply:
Regarding the Pyspelling dependencies, it uses either [Aspell](http://aspell.net/) or [Hunspell](https://hunspell.github.io/). I used Aspell for this project. See the Pyspelling [Setup & Overview](https://facelessuser.github.io/pyspelling/) page for more information.

Regarding the OS commands, I found the information online, but I can’t retrieve the sources because I cleared my browser history. Unfortunately, I’ve only been able to test them on my Mac M2 so far.